code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_node.py

```diff
@@ -0,0 +1,235 @@
+_PERIOD_CYCLE_STR: int = 500#250 # Period in ms of the storage node
+_PERIOD_CYCLE_MEAS: int = 400#140 # Period in ms of the measurement node
+_PERIOD_CYCLE_MAN: int = 1000 # Period in ms of the manager node
+
+class AppManNodeC(SysShdNodeC):
+    """Generate a classman node for the application .
+    """
+
+    def __init__(self, cs_id: int, working_flag: Event,
+                cycle_period: int= _PERIOD_CYCLE_MAN) -> None:
+        super().__init__(name= "App_Manager", cycle_period= cycle_period,
+                         working_flag=working_flag)
+        # Initialize attributes
+        self.cs_id: int = cs_id
+
+        # Initialize system structure except meas node
+        self.init_system()
+
+        signal(SIGINT, self.signal_handler)
+
+        log.info(f"{self.th_name} node initiliazed")
+
+    def signal_handler(self, sig, frame):
+        """Stop the keyboard .
+        """
+        log.critical('You pressed Ctrl+C! Stopping all threads...')
+        self.stop()
+        # TODO: raise alarm if stops node
+
+
+    def init_system(self) -> None:
+        """Initialize the system for this system
+        """
+        ### 1.0 Threads initialization ###
+        self.th_name = current_thread().name
+        ### 2.1 Manager thread ###
+        log.info(f"Starting {self.th_name} node ...")
+        self.working_app = Event()
+        self.working_app.set()
+
+        self.working_str = Event()
+        self.working_str.set()
+
+        self.working_meas = Event()
+        self.working_meas.set()
+
+        ### Shared objects and channels ###
+        self.__shd_gen_meas = SysShdSharedObjC(CyclerDataGenMeasC())
+        self.__shd_ext_meas = SysShdSharedObjC(CyclerDataExtMeasC())
+        self.__shd_all_status  = SysShdSharedObjC(CyclerDataAllStatusC())
+        __chan_alarms = SysShdChanC()
+        __chan_str_reqs = SysShdChanC()
+        __chan_str_data = SysShdChanC()
+        self.__shared_tags: CyclerDataMergeTagsC = CyclerDataMergeTagsC(status_attrs= [],
+                                                                 gen_meas_attrs= ['instr_id'],
+                                                                 ext_meas_attrs= [])
+
+        ### 1.1 Store thread ###
+        self._th_str = MidStrNodeC(name= 'MID_STR', working_flag= self.working_str,
+                shared_gen_meas= self.__shd_gen_meas, shared_ext_meas= self.__shd_ext_meas,
+                shared_status= self.__shd_all_status, str_reqs= __chan_str_reqs,
+                str_data= __chan_str_data, str_alarms= __chan_alarms,
+                cycle_period= _PERIOD_CYCLE_STR, cycler_station= self.cs_id,
+                cred_file= 'devops/.cred.yaml')
+        self._th_str.start()
+        # Get info from the cycler station to know which devices are compatible
+        cs_info = self.configure_cs(reqs_chan= __chan_str_reqs, data_chan= __chan_str_data)
+        for dev in cs_info.devices:
```

Comment:
Remove loop from here. It could make sense on mid_dabs or mid_meas

Reply:
It is for debug purpose

Reply:
It is for debug purpose